metricflow/model/dbt_transformations/dbt_metric_to_dimensions_rules.py

```diff
@@ -0,0 +1,125 @@
+                try:
+                    assert_metric_model_name(metric=metric)
+                    for dimension in metric.dimensions:
+                        objects.dimensions[metric.model.name][dimension] = Dimension(
+                            name=dimension,
+                            type=DimensionType.CATEGORICAL,
+                        ).dict()
+
+                except Exception as e:
+                    issues.append(
+                        ValidationError(message=str(e), extra_detail="".join(traceback.format_tb(e.__traceback__)))
+                    )
+
+        return ModelValidationResults.from_issues_sequence(issues=issues)
+
+
+class DbtTimestampToDimension(DbtTransformRule):
+    """Rule for mapping dbt metric timestamps to data source dimensions"""
+
+    @staticmethod
+    def run(dbt_metrics: Tuple[MetricNode, ...], objects: DbtTransformedObjects) -> ModelValidationResults:  # noqa: D
+        issues: List[ValidationIssue] = []
+        for metric in dbt_metrics:
+            # Creating dimensions only matters if their is a data source (model) to attach them too
+            if metric.model:
+                try:
+                    assert_metric_model_name(metric=metric)
+                    assert (
+                        metric.timestamp is not None
+                    ), f"Expected a value for `{metric.name}` metric's `timestamp`, got `None`"
+                    objects.dimensions[metric.model.name][metric.timestamp] = Dimension(
+                        name=metric.timestamp,
+                        type=DimensionType.TIME,
+                        type_params=DimensionTypeParams(is_primary=False, time_granularity=TimeGranularity.DAY),
+                    ).dict()
+
+                except Exception as e:
+                    issues.append(
+                        ValidationError(message=str(e), extra_detail="".join(traceback.format_tb(e.__traceback__)))
+                    )
+
+        return ModelValidationResults.from_issues_sequence(issues=issues)
+
+
+class DbtFiltersToDimensions(DbtTransformRule):
+    """Rule for mapping dbt metric filters to data source dimensions"""
+
+    @staticmethod
+    def _filter_value_is_timestamp(filter_value: str) -> bool:
+        """Determines if a given dbt MetricFilter represents a TIME type dimension or not
+
+        Every dbt MetricFilter represents a dimension. However, we aren't given
+        direct information if should be a CATEGORICAL or TIME dimension. The
+        good news is that every filter includes a `value` to compare with. Thus
+        we can check if this is iso datetime parseable to see if it should be
+        considered a TIME type dimension
+        """
+        try:
+            # filter values which represent date/time of some sort must start
+            # and end with single quotes, so we must remove these
+            potential_timestamp = filter_value.strip("'")
```

Comment:
From my reading of their docs, the `value` of a filter is always a direct scalar value, and cannot be a reference to another column. I could be wrong about that though, so I'll double check 🙃 additionally even if it can't right now, there's no guarantee it won't be able to do so in the future.

Reply:
So I asked dbt about this. They don't know. Whether or not it is possible now, I think I know how to identify when the `value` of a filter is a `field` itself. But we should gain clarity before implementing a solution.

Reply:
Resolved by: https://github.com/transform-data/metricflow/pull/324/commits/b2362f14e8055351379de30381b67ad391ca5f6e